d3ploy/demand_driven_deployment_inst.py

```diff
@@ -194,22 +201,37 @@ def enter_notify(self):
                 self.facility_constraintcommod,
                 self.facility_constraintval)
             commod_list = list(self.commodity_dict.keys())
+            for commod in commod_list:
+                self.installed_capacity[commod] = defaultdict(float)
+                self.installed_capacity[commod][0] = 0.
             for key, val in self.commodity_dict.items():
                 for key2, val2 in val.items():
                     if val2['constraint_commod'] != '0':
                         commod_list.append(val2['constraint_commod'])
-            commod_list = list(set(commod_list))
+            self.commod_list = list(set(commod_list))
             self.buffer_dict = di.build_buffer_dict(self.supply_buffer,
-                                                    commod_list)
+                                                    self.commod_list)
             self.buffer_type_dict = di.build_buffer_type_dict(
                 self.buffer_type, commod_list)
-            for commod in commod_list:
+            for commod in self.commod_list:
                 lib.TIME_SERIES_LISTENERS["supply" +
                                           commod].append(self.extract_supply)
                 lib.TIME_SERIES_LISTENERS["demand" +
                                           commod].append(self.extract_demand)
                 self.commodity_supply[commod] = defaultdict(float)
                 self.commodity_demand[commod] = defaultdict(float)
+            for child in self.children:
+                count = 0
+                for key, val in self.commodity_dict.items():
+                    for key2, val2 in val.items():
+                        if key2 == child.prototype:
+                            count = 1
```

Comment:
Are we sure we don't add to the count here? It seems like the count should ideally take on more than just a value of 1 or 0.... 

That is, I think you're aiming for:

```suggestion
                            count += 1
```

Reply:
I realize that this exception is not required because there is a built in exception in cyclus when using initialfacilitylist